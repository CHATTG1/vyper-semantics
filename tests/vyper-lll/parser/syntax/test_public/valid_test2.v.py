x: public(num(wei / sec))
y: public(num(wei / sec ** 2))
z: public(num(sec ** -1))

@public
def foo() -> num(sec ** 2):
    return self.x / self.y / self.z
