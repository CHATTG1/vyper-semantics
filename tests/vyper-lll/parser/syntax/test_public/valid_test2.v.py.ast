%pgm(,
  %svdecl(x, %unitT(%num, %udiv(%wei, %sec), false), %public)
  %svdecl(y, %unitT(%num, %udiv(%wei, %upow(%sec, 2)), false), %public)
  %svdecl(z, %unitT(%num, %upow(%sec, -1), false), %public),
  %fdecl(%@public, foo, , %unitT(%num, %upow(%sec, 2), false),
    %return(%binop(/, %binop(/, %svar(x), %svar(y)), %svar(z))))
)
