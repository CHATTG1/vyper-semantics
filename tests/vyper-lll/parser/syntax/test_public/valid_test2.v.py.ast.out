$%seq ( $return ( 0 , $lll ( $%seq ( $%seq ( $mstore ( 28 , $calldataload ( 0 ) ) , $mstore ( 32 , 1461501637330902918203684832716283019655932542976 ) , $mstore ( 64 , 170141183460469231731687303715884105727 ) , $mstore ( 96 , -170141183460469231731687303715884105728 ) , $mstore ( 128 , 1701411834604692317316873037158841057270000000000 ) , $mstore ( 160 , -1701411834604692317316873037158841057280000000000 ) ) , $if ( $eq ( $mload ( 0 ) , 3264763256 ) , $%seq ( $pass , $assert ( $iszero ( $callvalue ) ) , $%seq ( $mstore ( 0 , $clamp ( $mload ( 96 ) , $sdiv ( $clamp ( $mload ( 96 ) , $sdiv ( $sload ( 0 ) , $clamp_nonzero ( $sload ( 1 ) ) ) , $mload ( 64 ) ) , $clamp_nonzero ( $sload ( 2 ) ) ) , $mload ( 64 ) ) ) , $return ( 0 , 32 ) ) , $stop ) ) , $if ( $eq ( $mload ( 0 ) , 942212362 ) , $%seq ( $pass , $assert ( $iszero ( $callvalue ) ) , $%seq ( $mstore ( 0 , $sload ( 0 ) ) , $return ( 0 , 32 ) ) , $stop ) ) , $if ( $eq ( $mload ( 0 ) , 685153389 ) , $%seq ( $pass , $assert ( $iszero ( $callvalue ) ) , $%seq ( $mstore ( 0 , $sload ( 1 ) ) , $return ( 0 , 32 ) ) , $stop ) ) , $if ( $eq ( $mload ( 0 ) , 1223850360 ) , $%seq ( $pass , $assert ( $iszero ( $callvalue ) ) , $%seq ( $mstore ( 0 , $sload ( 2 ) ) , $return ( 0 , 32 ) ) , $stop ) ) ) , 0 ) ) )
